ResearchKit/Common/ORKHelpers.h

```diff
@@ -72,8 +72,11 @@
 #define ORK_DECODE_INTEGER(d,x) _ ## x = [d decodeIntegerForKey:@STRINGIFY(x)]
 #define ORK_ENCODE_INTEGER(c,x) [c encodeInteger:_ ## x forKey:@STRINGIFY(x)]
 
-#define ORK_ENCODE_UINT32(c,x) [c encodeObject:[NSNumber numberWithUnsignedLongLong:_ ## x] forKey:@STRINGIFY(x)]
-#define ORK_DECODE_UINT32(d,x) _ ## x = (uint32_t)[(NSNumber *)[d decodeObjectForKey:@STRINGIFY(x)] unsignedLongValue]
+#define ORK_ENCODE_UINT32(c,x) [c encodeInt32:_ ## x forKey:@STRINGIFY(x)]
+#define ORK_DECODE_UINT32(d,x) _ ## x = [d decodeInt32ForKey:@STRINGIFY(x)]
```

Comment:
I'm hesitant without a test verifying that the game result that uses this works correctly for a seed of 0xffffffff. Please add a sanity check for that serialization / deserialization  using NSKeyedArchiver to the tests. It's best to add to the ResearchKit unit tests directly, even though most of the serialization tests are actually in the ORKTest project.